runner.go

```diff
@@ -0,0 +1,67 @@
+package server
+
+import (
+	"net/http"
+	"net/url"
+	"time"
+)
+
+type Output struct {
+	Report   Report
+	Metadata struct {
+		Config     Config
+		FinishedAt time.Time
+	}
+}
+
+// Report represents the collected results of a benchmark test.
+type Report struct {
+	Records  []Record      `json:"records" firestore:"records"`
+	Length   int           `json:"length" firestore:"length"`
+	Success  int           `json:"success" firestore:"success"`
+	Fail     int           `json:"fail" firestore:"fail"`
+	Duration time.Duration `json:"duration" firestore:"duration"`
+}
+
+// Record represents the summary of a HTTP response.
+type Record struct {
+	Time   time.Duration `json:"time" firestore:"time"`
+	Code   int           `json:"code" firestore:"code"`
+	Bytes  int           `json:"bytes" firestore:"bytes"`
+	Error  string        `json:"error,omitempty" firestore:"error,omitempty"`
+	Events []Event       `json:"events" firestore:"events"`
+}
+
+// Event is a stage of an outgoing HTTP request associated with a timestamp.
+type Event struct {
+	Name string        `json:"name" firestore:"name"`
+	Time time.Duration `json:"time" firestore:"time"`
+}
+
+type Config struct {
+	Request Request
+	Runner  Runner
+	Output  struct {
+		Out      []string
+		Silent   bool
+		Template string
+	}
+}
+
+type Request struct {
+	Method string
+	URL    *url.URL
+	Header http.Header
+	Body   struct {
+		Type    string
+		Content []byte
+	}
+}
+
+type Runner struct {
+	Requests       int
+	Concurrency    int
+	Interval       time.Duration
+	RequestTimeout time.Duration
+	GlobalTimeout  time.Duration
+}
```

Comment:
The mega nesting it would produce is very noisy too and much less readable 😕
Would not exporting them be better ? I guess `gob` won't complain right ?

Reply:
I don't necessarily agree with the readability loss, as it also avoids having to navigate through the file to grasp the final structure, but anyway readability is not really a concern here because it's only a receiver structure for decoding the input request and store it as is.
Not exporting them is enough to me (and `gob` shouldn't complain about that as long as the fields themselves remain exported).